docs/benchmarkdotnet.md

```diff
@@ -0,0 +1,332 @@
+* it consumes the benchmark result to avoid dead code elimination
+* it prevents from inlining of the benchmark by wrapping it with a delegate
+* it prints the results to the console in GitHub markdown, so you can just copy-paste the printed table to GitHub
+* it exports the results to `.\BenchmarkDotNet.Artifacts\results` so you can store them for later use.
+
+A few useful links for you:
+
+* If you want to know more about BenchmarkDotNet features, check out the [Overview Page](http://benchmarkdotnet.org/Overview.htm).
+* If you want to use BenchmarkDotNet for the first time, the [Getting Started](http://benchmarkdotnet.org/GettingStarted.htm) will help you.
+* If you want to ask a quick question or discuss performance topics, use the [gitter](https://gitter.im/dotnet/BenchmarkDotNet) channel.
+
+## Running the Benchmarks
+
+### Prerequisites
+
+To run the benchmarks, you need to [download dotnet cli](./prerequisites.md) first.
+
+### Interactive Mode
+
+To run the benchmarks in interactive mode you have to execute `dotnet run -f $targetFrameworkMoniker` in the folder with benchmarks project.
+
+```cmd
+C:\Projects\performance\src\benchmarks\micro> dotnet run -f netcoreapp3.0
+Available Benchmarks:
+  #0   Burgers
+  #1   ByteMark
+  #2   CscBench
+  #3   LinqBenchmarks
+  #4   SeekUnroll
+  #5   Binary_FromStream<LoginViewModel>
+  #6   Binary_FromStream<Location>
+  #7   Binary_FromStream<IndexViewModel>
+  #8   Binary_FromStream<MyEventsListerViewModel>
+  #9   Binary_FromStream<CollectionsOfPrimitives>
+  #10  Binary_ToStream<LoginViewModel>
+  ..... // the list continues
+
+You should select the target benchmark(s). Please, print a number of a benchmark (e.g. '0') or a contained benchmark caption (e.g. 'Burgers'):
+```
+
+And select one of the benchmarks from the list by either entering its number or name.
+
+### Command Line
+
+#### Filtering the Benchmarks
+
+You can filter the benchmarks using `--filter $globPattern` console line argument.
+
+The glob patterns are applied to full benchmark name: namespace.typeName.methodName. Examples:
+
+* `dotnet run -f netcoreapp3.0 --filter BenchmarksGame*` - will run all the benchmarks from BenchmarksGame namespace
+* `dotnet run -f netcoreapp3.0 --filter *.Richards.*` - will run all the benchmarks with type name Richards
+* `dotnet run -f netcoreapp3.0 --filter *.ToStream` - will run all the benchmarks with method name ToStream
+* `dotnet run -f netcoreapp3.0 --filter *` - **will run ALL benchmarks**
+
+You can provide many filters (logical disjunction):
+
+```cmd
+dotnet run -f netcoreapp3.0 --filter System.Collections*.Dictionary* *.Perf_Dictionary.*
+```
+
+To print a **joined summary** for all of the benchmarks (by default printed per type), use `--join`:
+
+```cmd
+dotnet run -f netcoreapp2.1 --filter BenchmarksGame* --join
+```
+
+Please remember that on **Unix** systems `*` is resolved to all files in current directory, so you need to escape it `'*'`. The filter is **case insensitive**.
```

Comment:
nit: "The filter is **case insensitive**." could be mentioned above in line 87.

Reply:
good point! it's important and should be mentioned first